### PR TITLE
Add ansible epoch.fact

### PR DESCRIPTION
**Pull request by dincho**
example output:
```
ae-dev1-epoch-n0 | SUCCESS => {
    "ansible_facts": {
        "ansible_local": {
            "epoch": {
                "config": {
                    "chain": {
                        "db_path": "./db48",
                        "persist": true
                    },
                    "http": {
                        "debug": false,
                        "external": {
                            "port": 3013
                        },
                        "internal": {
                            "listen_address": "127.0.0.1",
                            "port": 3113
                        }
                    },
                    "keys": {
                        "dir": "keys",
                        "password": "secret"
                    },
                    "logging": {
                        "level": "debug"
                    },
                    "metrics": {
                        "host": "127.0.0.1",
                        "port": 8125
                    },
                    "mining": {
                        "autostart": true,
                        "cuckoo": {
                            "miner": {
                                "executable": "mean28s-generic",
                                "extra_args": "-t 5",
                                "node_bits": 28
                            }
                        },
                        "expected_mine_rate": 300000
                    },
                    "peers": [
                        "aenode://pp$2JVzwdAMEzz7MjPdPtfCHLotSFTJxMHRos4Yhj1iNYj58f3wLs@31.13.249.75:3015"
                    ],
                    "sync": {
                        "port": 3015
                    },
                    "websocket": {
                        "internal": {
                            "listen_address": "127.0.0.1",
                            "port": 3114
                        }
                    }
                },
                "peer_key": "pp$2ZCqndybQ67RpTQxnQtzzzswoeb2VoTFwcc9RB6Q9djwpiSyFD",
                "revision": "e5cb466452a872e44825c130766e2ddcf5fea370",
                "version": "0.12.0"
            }
        }
    },
    "changed": false
}
```

https://www.pivotaltracker.com/story/show/156535422

### CODE REVIEW

**Comment by dincho** [comment]
I don't think that CI failure is related. Reproduced locally. I guess it's because of the recent OS update.

**Comment by dincho** [comment]
I fixed the CI fail in this PR.

File: ansible/setup.yml [hunk 1]
```diff
@@ -81,9 +81,21 @@
...
+        - python-yaml
       tags:
         - dev-tools
 
+    - name: "Create custom fact directory"
+      file:
+        path: "/etc/ansible/facts.d"
+        state: "directory"
+
+    - name: "Install epoch fact file"
```
Comment (lucafavatella) [inline ansible/setup.yml]: I see you do not include
```
- name: "Re-run setup to use custom facts"
  setup: ~
```
(from https://medium.com/@jezhalford/ansible-custom-facts-1e1d1bf65db8 linked in https://www.pivotaltracker.com/n/projects/2152594/stories/156535422 )

as these facts are are meant to be used by subsequent PR amending playbook in https://github.com/aeternity/epoch , right?

File: ansible/files/epoch.fact [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import os
+import subprocess
+import yaml
+import json
+
+node_path = "/home/epoch/node"
```
Comment (lucafavatella) [inline ansible/files/epoch.fact]: I see this is duplicate of https://github.com/aeternity/epoch/blob/e72c0e2128eb82e61b80afa85d159f34cf489c4f/deployment/ansible/deploy.yml#L43

ok.

File: ansible/setup.yml [hunk 3]
```diff
@@ -81,9 +81,21 @@
...
+      copy:
+        src: files/epoch.fact
+        dest: /etc/ansible/facts.d/epoch.fact
```
Comment (lucafavatella) [inline ansible/setup.yml]: This file seems coupling infrastructure with deployed app, though I see it is in /etc so kind of part of infrastructure.

Ok.

File: ansible/setup.yml (hunk 1 above)
Comment (dincho) [reply ansible/setup.yml]: Yes, we don't need the new facts in this playbook.

**Pull request merged**